sdk/python/kfp/compiler/_component_builder.py

```diff
@@ -312,9 +342,16 @@ def _generate_entrypoint(self, component_func):
     codegen.writeline(call_component_func)
 
     # Serialize output
-    codegen.writeline('from pathlib import Path')
-    codegen.writeline('Path(_output_file).parent.mkdir(parents=True, exist_ok=True)')
-    codegen.writeline('Path(_output_file).write_text(str(output))')
+    if python_version == 'python3':
```

Comment:
why not just keep the python2 code which works in both versions?

Reply:
The codes that work for python3 have been in the code repo for some time and are working fine without reported bugs. 
Since the python2 code works fine in the sample tests. I will remove the python3 related codes here.